airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/buffers/StreamAwareQueue.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination_async.buffers;
+
+import io.airbyte.integrations.destination_async.partial_messages.PartialAirbyteMessage;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class StreamAwareQueue {
```

Comment:
i remember we said this abstraction has low value. do we still want to keep it? or should we get rid of it?

Reply:
i'm open to getting rid of it. i would say, if we want to reuse the `MemoryBoundedLinkedBlockingQueue` in platform for the buffer project, then maybe we should keep this abstraction so that we can share just `MemoryBoundedLinkedBlockingQueue` and not other info about the destination with the platform. if we aren't going to reuse it, then i think we can get rid of it. either way, would rather do it in a later PR just to get this one cleared. lmk if you have a preference one way or another as this affects move and @gosusnp .

Reply:
I would not plan on sharing this with platform at the moment:
* when testing the buffer project, I see a significant performance improvement by using a ConcurrentQueue rather than a BlockingQueue
* Since the repo split, did we land on a strategy for sharing code between connector/platform?

Reply:
> Since the repo split, did we land on a strategy for sharing code between connector/platform?

lol nope. and sounds like we don't have to yet. 😛 

> when testing the buffer project, I see a significant performance improvement by using a ConcurrentQueue rather than a BlockingQueue

nice!